### DISCUSSION TITLE
Supabase realtime updates

### DISCUSSION THREAD

**Discussion by TomasRences** [supabase/supabase#14104] [Questions] [answered]
Hello community! My name is Tomas and our company started use supabase recently for our new project. As frontent we use Angular (v15) and we want to use a realtime feature in our application.
But we face this issues:

**1.)** Our application use the NGRX store and effects for managing application state.
So after loading data from supabase, we save entities in application store and render store content in our components.
In requests we join nested objects in select methods:

```
// objects.service.ts
const queryText = `
  *,
  status:statuses !inner(
    *
  ),
  priority:priorities!inner(
    *
  )
`;

get(filters?: Filter[], range?: DataRange | undefined): Observable<any> {
  let query = this.supabaseService.supabase.from('objects').select(
    queryText
  )

  query = buildFilter(query, filters, range);

  return of(query).pipe(
    switchMap(data => data)
  )
}
```

"get" function is called in NGRX Effect and successful result is pushed into store.

We want to implement realtime data updates for all co-workers so when some change event occurs, changes are also pushed to the NGRX store.
The Problem we have is, "new" objects in change event are without nested data ("status", "priority" in example above), is there any way to get nested joined data in change event?
Currently we are reloading data after update and it works, but it's an another request to server.

```
// app.component.ts
this.supabase.supabase
      .channel('any')
      .on('postgres_changes', { event: '*', schema: '*' }, payload => {
        console.log(payload);

        switch (payload.eventType) {
          case "INSERT":
          case "UPDATE": {
            if (payload.table === 'sokoObjects') {

              // request to supabase
              this.store.dispatch(LoadObjectsRequest({
                payload: {
                  filters: [{
                    column: 'id',
                    operator: 'in',
                    value: [payload.new["id"]]
                  }]
                }
              }));

            }

            break;
          }
          case "DELETE": {
            // remove object from store

            if (payload.table === 'sokoObjects') {
              this.store.dispatch(DeleteObjectsSuccess({
                payload: {
                  ids: [payload.old["id"]]
                }
              }));
            }

            break;
          }
        }
      }).subscribe();
```

**2.)** In batch update of many objects, all changes are sent as separate messages:

![image](https://user-images.githubusercontent.com/22801505/235931699-efa99ff8-91f7-40f8-a7af-3dadf3a64c1a.png)

Is it possible to have bulk messages with many changed rows in array?

Hope this is not a duplicate post and thank you for any suggestion.

**Comment by GaryAustin1** [comment, marked as answer]
Those cases are not currently supported.  The join feature has been mentioned and requested several times.

https://github.com/supabase/realtime/issues/303

I believe I have seen the bulk messages feature requested also, but did not search for it in the realtime github I just linked.
